### ISSUE TITLE
When multiple users are interacting with the same saved search it can result in a permissions error.

### ISSUE THREAD

**Issue by linuspahl** [Graylog2/graylog2-server#12805]
## Description

When multiple users are interacting with and saving the same saved search it can result in an error.

Imagine the following use case
1. User A creates a saved search and shares it with User B.
2. User B is opening the saved search
3. User A is e.g. changing the search query and saving the changes
4. User B is not reloading the saved search (since step 2) and just reexecutes it. 

User A and User B both have the Views Manager and User Inspector, but not he admin role.
In this case User B sees the following error:
```
The permissions check for the following request failed,
while trying to access /search/[search-id].
```

## Your Environment

* Graylog Version: 4.3

**Comment by luk-kaminski** [comment]
@kmerz :

After step 3 the saved search is associated with a new search document with a new ID in our Mongo DB. 
In step 4 FE invokes `execute` endpoint, using old search document ID, which was associated with saved search before it was changed in step 3.

The old search document has no reference to saved search, therefore permission check in `SearchDomain` class throws an exception.

It is a FE bug. 
Imho when working with saved searches FE needs to verify if saved search has not been changed before executing them.
It is a business decision what to do if that change is discovered: reload fresh version of the search (with some notification about it) or let the user work on an old version (also telling them that their searches are based on a stale saved search and probably blocking the possibility to save their changes, as merging saved searches is probably not smth we would like to touch).

If we are interested in solving only permission problem, we could do it in BE using some hackish solution (i.e. saved search stores previous associated searches IDs and we use it in permission check), but other related problems will still be there...

**Comment by kmerz** [comment]
@luk-kaminski I agree this is a product decision: 

The ideal solution would be to notify the user that the search they are using has been updated, and they need to reload if they want to see the updated version. But it would be really helpful to give the user the possibility to finish their current work at hand. We should not block User B from completing their job due to a permission error.

As a story:
1. User A creates a saved search and shares it with User B.
2. User B is opening the saved search
3. User A is e.g., changing the search query and saving the changes
4. User B is not reloading the saved search (since step 2) and executes it.
5. User B gets a Notification that the current search is outdated, and he gets the results of the search he executed.

**Comment by luk-kaminski** [comment]
@maxiadlovskii - is it a duplicate of what you have mentioned yesterday in the meeting?
